Glimmer packages

stefanpenner opened the pull request:
this is the "quickest" way to import glimmer into ember.... This is to allow work to begin, that being said. I plan to actually have them unify build-pipelines, as this should allow a much better development experience.

rwjblue committed:
Begin fleshing out addition of Glimmer (2.0) package for integration.

stefanpenner committed:
basic glimmer import (From output of build node_module)

I plan to still improve this, but this should help move stuff long.

stefanpenner commented:
this will basically keep failing until we publish glimmer with dist built etc.

rwjblue commented on ember-cli-build.js:
```diff
@@ -65,7 +74,17 @@ module.exports = function() {
+      'glimmer-util':          glimmerPackage('glimmer-util'),
+      'glimmer-wire-format':   glimmerPackage('glimmer-wire-format'),
+      'glimmer':               glimmerPackage('glimmer'),
+      'simple-html-tokenizer': glimmerPackage('simple-html-tokenizer')
```
This is duplicated. Do we need two different versions of SHT? If not, then removing the duplicate should be fine, if we do need different ones then we need to figure out how to namespace them to prevent clobbering.

stefanpenner replied:
> This is duplicated. Do we need two different versions of SHT? If not, then removing the duplicate should be fine, if we do need different ones then we need to figure out how to namespace them to prevent clobbering.

this is a @chancancode question :P

chancancode replied:
Looks like we did some import changes in SHT: https://github.com/tildeio/simple-html-tokenizer/commits/master (htmlbars uses c4787e3 and glimmer uses 8b09fc2). If we can change htmlbars to work with the new structure then we can just release a new SHT version and use that in htmlbars (and do another htmlbars release?).

chancancode replied:
Looking at the changes, it seems like it might Just Work™ in htmlbars too?

stefanpenner committed:
power of broccoli

* import glimmers build… use it

stefanpenner committed:
We currently consume the ES6 output of glimmer build (post TS etc). So we must transpile it.

If we restructure the glimmer build slightly, we could just consume the separate AMD modules (likely as one would expect..)

stefanpenner committed:
simpler glimmer-engine import

stefanpenner committed:
depend on relevant feature branch

stefanpenner commented:
oops

rwjblue committed:
Only load glimmer-engine if feature is enableable.

rwjblue committed:
Use master of glimmer for integration efforts.

We will lock this down when we have made more progress.

rwjblue committed:
Add ember-glimmer feature flag.

rwjblue commented:
OK, just pushed a few changes:

* Added a feature flag for `ember-glimmer`.
* Only added the glimmer-engine if the feature flag is either `null` or `true`.
* Update a few things in glimmer-engine to enable running the build without `npm link` (still pending in https://github.com/tildeio/glimmer/pull/49).

Once https://github.com/tildeio/glimmer/pull/49 is landed, this is ready to merge (after confirming Travis is passing).

chancancode commented on ember-cli-build.js:
```diff
@@ -42,7 +42,42 @@ function babelConfigFor(environment) {
+        'amd/glimmer-compiler.amd.js',
+        /*'amd/glimmer-runtime.amd.js', */
+        'amd/tests.amd.js'
+      ]
```
This should be:

```js
'amd/glimmer-common.amd.js', // stuff shared between runtime and compiler, e.g. utils
'amd/glimmer-compiler.amd.js', // for "server compilation", depends on common
'amd/glimmer-runtime.amd.js', // runtime stuff, depends on common
'amd/glimmer-tests.amd.js' // tests, depends on common, compiler and runtime
```

stefanpenner commented:
LGTM

Commit:
Pull in all glimmer packages

Commit:
Pull in glimmer sourcemap files

Commit:
Include a stub for the tests directory to keep the watcher happy

chancancode committed:
Bump ember-cli-built to fix IE issues

This version adds the `proto-to-assign` transform which makes static
methods work on IE <= 10

chancancode committed:
Add a dummy test to keep the test runner happy

rwjblue commented:
Restarted sauce labs tests...

chancancode committed:
Add a toggle for hidding skipped tests
